sdk/java/src/test/java/com/chain/integration/UpdateTagsTest.java

```diff
@@ -0,0 +1,199 @@
+package com.chain.integration;
+
+import com.chain.TestUtils;
+import com.chain.api.*;
+import com.chain.exception.*;
+import com.chain.http.*;
+
+import org.junit.Test;
+
+import java.util.*;
+
+import static org.junit.Assert.*;
+
+public class UpdateTagsTest {
+  @Test
+  public void accountTags() throws Exception {
+    Client client = TestUtils.generateClient();
+    MockHsm.Key key = MockHsm.Key.create(client);
+
+    Account account1 =
+        new Account.Builder().addRootXpub(key.xpub).setQuorum(1).addTag("x", "zero").create(client);
+    Account account2 =
+        new Account.Builder().addRootXpub(key.xpub).setQuorum(1).addTag("y", "zero").create(client);
+
+    Map<String, Object> update1, update2;
+
+    // Account tag update
+
+    update1 = new HashMap<>();
+    update1.put("x", "one");
+
+    new Account.TagUpdateBuilder().forId(account1.id).setTags(update1).update(client);
+
+    Account.Items accounts =
+        new Account.QueryBuilder()
+            .setFilter("id=$1")
+            .addFilterParameter(account1.id)
+            .execute(client);
+
+    while (accounts.hasNext()) {
+      assertEquals(accounts.next().tags.get("x"), "one");
+    }
+
+    // Account tag update that raises an error
+
+    try {
+      update1 = new HashMap<>();
+      update1.put("x", "two");
+
+      new Account.TagUpdateBuilder()
+          // ID intentionally omitted
+          .setTags(update1)
+          .update(client);
+    } catch (APIException e) {
+      assertTrue(e.toString().contains("CH051"));
+    }
+
+    // Account tag batch update
+
+    update1 = new HashMap<>();
+    update1.put("x", "three");
+    update2 = new HashMap<>();
+    update2.put("y", "three");
+
+    BatchResponse<SuccessMessage> batch =
+        Account.updateTagsBatch(
+            client,
+            Arrays.asList(
+                new Account.TagUpdateBuilder().forId(account1.id).setTags(update1),
+                new Account.TagUpdateBuilder().forId(account2.id).setTags(update2)));
+
+    assertEquals(batch.errorsByIndex().size(), 0);
+
+    accounts =
+        new Account.QueryBuilder()
+            .setFilter("id=$1 OR id=$2")
+            .addFilterParameter(account1.id)
+            .addFilterParameter(account2.id)
+            .execute(client);
+
+    List<Account> accountList = new ArrayList<>();
+    while (accounts.hasNext()) {
+      accountList.add(accounts.next());
+    }
+
+    assertEquals(accountList.get(0).tags.get("y"), "three");
+    assertEquals(accountList.get(1).tags.get("x"), "three");
```

Comment:
nit: can we add a comment about the query results being in reverse chron. order of creation? I first questioned if we could just rely on this ordering until I checked.